src/shared/conf-parser.c

```diff
@@ -320,8 +325,14 @@ int config_parse(const char *unit,
                 bool escaped = false;
 
                 if (!fgets(buf, sizeof buf, f)) {
```

Comment:
hmm, i'd probably just have implemented getline_with_limit() or so, which is exactly like getline() and takes a size limit. By doing manual locking we should be able to implement this as efficiently outside of libc as inside of libc and it would be useful at various places...